src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Avx512F.cs

```diff
@@ -639,6 +845,119 @@ public abstract class VL
             ///   VPSRAVQ ymm1 {k1}{z}, ymm2, ymm3/m256/m64bcst
             /// </summary>
             public static Vector256<long> ShiftRightArithmeticVariable(Vector256<long> value, Vector256<ulong> count) => ShiftRightArithmeticVariable(value, count);
+
+            /// <summary>
+            /// __m128i _mm_ternarylogic_si128 (__m128i a, __m128i b, __m128i c, byte imm)
+            ///   VPTERNLOGD xmm1 {k1}{z}, xmm2, xmm3/m128, imm8
+            /// The above native signature does not exist. We provide this additional overload for consistency with the other bitwise APIs.
+            /// </summary>
+            public static Vector128<sbyte> TernaryLogic(Vector128<sbyte> a, Vector128<sbyte> b, Vector128<sbyte> c, [ConstantExpected] byte control) => TernaryLogic(a, b, c, control);
```

Comment:
What's `[ConstantExpected]` by the way and do we switch-expand if it's called with a variable?

Reply:
It's an analyzer that flags if users aren't passing in a constant (or a value that is itself marked `[ConstantExpected]`).  It's not perfect, but it gets most of the job done and flags the obvious cases where we'll have to pessimize.

Yes, we fallback to the recursive call and switch expand in that. Fixing that to be done late phase (such as lowering) is desirable but requires various ABI fixes so the call is happy with the SIMD args again.